resolver.go

```diff
@@ -184,7 +189,14 @@ func createRespMsg(query *dns.Msg) *dns.Msg {
 }
 
 func (r *resolver) handleIPQuery(name string, query *dns.Msg, ipType int) (*dns.Msg, error) {
-	addr, ipv6Miss := r.sb.ResolveName(name, ipType)
+	var addr []net.IP
+	var ipv6Miss bool
+	if r.sb != nil {
+		addr, ipv6Miss = r.sb.ResolveName(name, ipType)
+	} else if r.network != nil {
```

Comment:
We can still avoid these boilerplate if/else checks in multiple places by moving the lookup functions ResolveName, ResolveIP, ResolveService to a new interface in resolver.go. These methods can be implemented by both Sandbox and Network objects. And `resolver.sb` can be changed to `resolver.backend` which can be either sandbox or the network object. This will make the resolver agnostic to how the backend lookup functions are implemented.